test/core/internals.js

```diff
@@ -94,15 +94,20 @@ describe('Core htmx internals Tests', function() {
     htmx._('shouldCancel')({ type: 'click' }, anchorThatShouldNotCancel).should.equal(false)
 
     var form = make('<form></form>')
-    htmx._('shouldCancel')({ type: 'submit' }, form).should.equal(true)
+    htmx._('shouldCancel')({ type: 'submit', target: document.createElement('form') }, form).should.equal(true)
+
+    htmx._('shouldCancel')({ type: 'click', target: document.createElement('form') }, form).should.equal(true)
+
+    // function works also when target isn't an Element
+    htmx._('shouldCancel')({ type: 'click', target: null }, form).should.equal(false)
 
     var form = make("<form><input id='i1' type='submit'></form>")
     var input = byId('i1')
-    htmx._('shouldCancel')({ type: 'click' }, input).should.equal(true)
+    htmx._('shouldCancel')({ type: 'click', target: document.createElement('input') }, input).should.equal(true)
```

Comment:
Shouldn't we use the `input` var here though instead of creating a dummy element on the fly? As `input` is the clicked element, the event's target would be that very element in a real situation right?

Same goes for the `form` above, and the `button` after

Reply:
Yes you're right